UdpCallbacks.cs

```diff
@@ -105,9 +105,17 @@ private bool ReceivePunchPackets(int traversalStart, int timeToSync)
             int rxCount = 0;
             do {
                 Console.Write (" <");
-                punch = udpc.Receive (ref ep);
-                rxCount += (ep.Equals(udpc.Client.RemoteEndPoint) && IsPunchPacket(punch, timeToSync)) ? 1:0;
-                PLog.DEBUG ("\nReceived pkt, endpoint now={0}, received size={1}", punch, punch.Length);
+                try
+                {
+                    punch = udpc.Receive (ref ep);
+                    rxCount += (ep.Equals(udpc.Client.RemoteEndPoint) && IsPunchPacket(punch, timeToSync)) ? 1:0;
+                    PLog.DEBUG ("\nReceived pkt, endpoint now={0}, received size={1}", punch, punch.Length);
+                }
+                catch (Exception e)

```

Comment:
what exception type are you really expecting here? should be better to be more specific with the type

Reply:
SocketException, updated the catch